types/consensus_data.go

```diff
@@ -46,6 +46,12 @@ func (c *Contributions) HashTreeRootWith(hh ssz.HashWalker) error {
 	return nil
 }
 
+// DataFetcher asynchronusly fetches data from the beacon node upon instantiation
+type DataFetcher struct {
```

Comment:
Why not interface? Using concrete types is not flexible and inconvenient for testing and when spec repo is used by other repositories

Reply:
@nkryuchkov @GalRogozinski 
Usually I support using concrete types. As they are more predictable than interfaces. but this time you are using this concrete type as an interface so it might make more sense for it to be an interface ( but then also moved to the runners ).
the rule is -
Return concrete types
Receive interfaces